akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala

```diff
@@ -670,9 +670,12 @@ private[remote] class ArteryTransport(_system: ExtendedActorSystem, _provider: R
         flushingPromise.future
       }
     implicit val ec = remoteDispatcher
-    flushing.recover { case _ ⇒ Done }.map { _ ⇒
-      killSwitch.shutdown()
 
+    for {
+      _ ← flushing.recover { case _ ⇒ Done }
+      _ = killSwitch.shutdown()
+      _ ← streamsCompleted
```

Comment:
I put the recover on the futures where failure would actually happen, this is the aggregate, I don't think Future.sequence will ever fail unless the sequenced failures does.